synapse/res/templates/sso_auth_account_details.html

```diff
@@ -130,15 +130,15 @@
   </head>
   <body>
     <header>
-      <h1>Your account is nearly ready</h1>
-      <p>Check your details before creating an account on {{ server_name }}</p>
+      <h1>Choose your user name</h1>
+      <p>This is required to create your account on {{ server_name }}, and you can't change this later.</p>
     </header>
     <main>
       <form method="post" class="form__input" id="form">
         <div class="username_input" id="username_input">
           <label for="field-username">Username</label>
           <div class="prefix">@</div>
-          <input type="text" name="username" id="field-username" autofocus>
+          <input type="text" name="username" id="field-username" value="{{ user_attributes.remote_user_id }}" autofocus>
```

Comment:
I'm a bit concerned that this might not be a sensible thing to do for non-Apple providers (often the remote_user_id is an entirely opaque string), so I think it would be better to make this optional.

Also: if your remote_user_id matches an MXID that is already registered (presumably via direct registration or a different SSO provider), we'll prepopulate the form with an invalid option, which is not great UX.

One way to do this: The `oidc_config` already has a field `localpart_template`, which defines a way to build a localpart from the remote details (and handles clashing MXIDs). However, the act of setting it means that the user is not given the opportunity to change the generated MXID. So, we could add another field to the config `confirm_generated_localpart: true|false` or somesuch, which means we'll show this page even if `localpart_template` is set.

Reply:
> often the remote_user_id is an entirely opaque string

Yes indeed, I've just realised I've misunderstood what it's meant to be.

Your suggestion looks sound, I'll go with that, thanks!